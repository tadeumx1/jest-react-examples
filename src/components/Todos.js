import React, { Component } from 'react';

import { connect } from 'react-redux';

class Todos extends Component {

  render() {
    return (
      <ul className="Todos">

        { this.props.todos.map(todo => (

            <li key={todo.id}>{todo.text}</li>

        )) }

      </ul>
    );
  }
}

// Utilizando a composição do React

const mapStateToProps = state => ({

    todos: state.todos,

});

export default connect(mapStateToProps)(Todos);
