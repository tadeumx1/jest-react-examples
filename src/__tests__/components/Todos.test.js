import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Todos from '../../components/Todos';

import configureStore from 'redux-mock-store';
const mockStore = configureStore();

const initialState = {

  todos: [

    { id: 0, text: 'Fazer café'},
    { id: 0, text: 'Acessar o slack'},
    { id: 0, text: 'Ver RocketLive'},

  ],

};

// Isso é o equivalente a declarar nossos Reducers

const store = mockStore(initialState);

describe('Testing Todos Component', () => {

  it('should render correctly', () => {

    const wrapper = shallow(

      <Todos />,

      // Passando store pelo contexto do React
      { context: { store }}

    ).dive();

    expect(wrapper).toMatchSnapshot();

  });

});
