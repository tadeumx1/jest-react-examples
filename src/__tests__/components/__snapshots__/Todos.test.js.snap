// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todos
    dispatch={[Function]}
    todos={
        Array [
            Object {
              "id": 0,
              "text": "Fazer café",
            },
            Object {
              "id": 0,
              "text": "Acessar o slack",
            },
            Object {
              "id": 0,
              "text": "Ver RocketLive",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          Fazer café
</li>,
        <li>
          Acessar o slack
</li>,
        <li>
          Ver RocketLive
</li>,
      ],
      "className": "Todos",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Fazer café",
        },
        "ref": null,
        "rendered": "Fazer café",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Acessar o slack",
        },
        "ref": null,
        "rendered": "Acessar o slack",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Ver RocketLive",
        },
        "ref": null,
        "rendered": "Ver RocketLive",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            Fazer café
</li>,
          <li>
            Acessar o slack
</li>,
          <li>
            Ver RocketLive
</li>,
        ],
        "className": "Todos",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Fazer café",
          },
          "ref": null,
          "rendered": "Fazer café",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Acessar o slack",
          },
          "ref": null,
          "rendered": "Acessar o slack",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Ver RocketLive",
          },
          "ref": null,
          "rendered": "Ver RocketLive",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;
